### PR TITLE
Implement SyncCommitteeContributionPool and build SyncAggregate for blocks

### LINKED ISSUE #3769: HF1 - Sync Committees - Block Proposals
### Description
Support block proposals including the new `SyncAggregate` from Altair. Block proposal section of https://github.com/ethereum/eth2.0-specs/blob/v1.1.0-alpha.2/specs/altair/validator.md#block-proposal

* Add support for `SignedContributionAndProof` gossip
* Create sync subcommittee contributions pool to track best contributions from each subcommittee
* Combine best sync subcommittee contributions into created blocks
    * Note that only contributions which point to blocks on the chain leading to the proposed block can be included

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Collect sync committee contributions from gossip and use them to create a `SyncAggregate` when producing Altair blocks.

## Fixed Issue(s)
fixes #3769 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/SyncCommitteeUtil.java [hunk 1]
```diff
@@ -254,4 +263,36 @@ public UInt64 getMinEpochForSyncCommitteeAssignments(final UInt64 requiredEpoch)
...
+    for (SyncCommitteeContribution contribution : contributions) {
+      final int subcommitteeIndex = contribution.getSubcommitteeIndex().intValue();
+      contribution
+          .getAggregationBits()
+          .streamAllSetBits()
+          .forEach(
+              index -> {
+                final int participantIndex =
+                    specConfig.getSyncCommitteeSize()
+                            / SYNC_COMMITTEE_SUBNET_COUNT
```
Comment (rolfyone) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/SyncCommitteeUtil.java]: call `getSubcommitteeSize` here?

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeContributionPool.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                betterContribution(existingContribution, contribution));
+  }
+
+  private SyncCommitteeContribution betterContribution(
+      final SyncCommitteeContribution a, SyncCommitteeContribution b) {
+    if (a == null) {
+      return b;
+    } else if (b == null) {
+      return a;
+    } else {
```
Comment (rolfyone) [inline ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeContributionPool.java:88]: biggest of nitty nits: don't really need the else, the final return could be top level

**Review by rolfyone — APPROVED** [review]
LGTM

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/SyncCommitteeUtil.java (hunk 1 above)
Comment (ajsutton) [reply ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/SyncCommitteeUtil.java]: Good spot.

File: ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeContributionPool.java (hunk 2 above)
Comment (ajsutton) [reply ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/synccommittee/SyncCommitteeContributionPool.java:88]: Yeah but technically you could skip all the `else` in here.  It's personal preference but I tend to like making it clear the three cases are linked with things like this.

**Pull request merged**